bundles/org.openhab.binding.tado/src/main/java/org/openhab/binding/tado/internal/TadoHvacChange.java

```diff
@@ -30,42 +32,47 @@
 import org.openhab.binding.tado.internal.builder.ZoneStateProvider;
 import org.openhab.binding.tado.internal.handler.TadoZoneHandler;
 import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.binding.ThingHandler;
 
 /**
  * Builder for incremental creation of zone overlays.
  *
  * @author Dennis Frommknecht - Initial contribution
  */
+@NonNullByDefault
 public class TadoHvacChange {
-    private TadoZoneHandler zoneHandler;
+
+    private final TadoZoneHandler zoneHandler;
+    private final TerminationConditionBuilder terminationConditionBuilder;
+    private final ZoneSettingsBuilder settingsBuilder;
 
     private boolean followSchedule = false;
-    private TerminationConditionBuilder terminationConditionBuilder;
-    private ZoneSettingsBuilder settingsBuilder;
 
     public TadoHvacChange(Thing zoneThing) {
-        if (!(zoneThing.getHandler() instanceof TadoZoneHandler)) {
+        ThingHandler handler = zoneThing.getHandler();
+        if (!(handler instanceof TadoZoneHandler)) {
             throw new IllegalArgumentException("TadoZoneThing expected, but instead got " + zoneThing);
         }
-
-        this.zoneHandler = (TadoZoneHandler) zoneThing.getHandler();
-        this.terminationConditionBuilder = TerminationConditionBuilder.of(zoneHandler);
-        this.settingsBuilder = ZoneSettingsBuilder.of(zoneHandler);
+        zoneHandler = (TadoZoneHandler) handler;
+        terminationConditionBuilder = TerminationConditionBuilder.of(zoneHandler);
+        settingsBuilder = ZoneSettingsBuilder.of(zoneHandler);
     }
 
     public TadoHvacChange withOperationMode(OperationMode operationMode) {
         switch (operationMode) {
             case SCHEDULE:
-                return followSchedule();
+                followSchedule();
+                break;
```

Comment:
Why these additional lines when builder pattern seems implemented and each method returning self?

Reply:
Good catch. Included in next commit.